plugins/woocommerce-admin/client/homescreen/mobile-app-modal/components/JetpackInstallationStepper.tsx

```diff
@@ -0,0 +1,129 @@
+/**
+ * External dependencies
+ */
+import React, { useState, useEffect } from '@wordpress/element';
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Spinner, Stepper, StepperProps } from '@woocommerce/components';
+
+/**
+ * Internal dependencies
+ */
+import {
+	SendMagicLinkButton,
+	useJetpackPluginState,
+	JetpackPluginStates,
+} from './';
+
+export const JetpackInstallationStepper = ( {
+	step,
+	sendMagicLinkHandler,
+}: {
+	step: 'first' | 'second';
+	sendMagicLinkHandler: () => void;
+} ) => {
+	const { installHandler, jetpackConnectionData, state } =
+		useJetpackPluginState();
+
+	const [ isWaitingForRedirect, setIsWaitingForRedirect ] = useState( false );
+
+	const [ stepsToDisplay, setStepsToDisplay ] = useState<
+		StepperProps[ 'steps' ] | undefined
+	>( undefined );
+	// we need to generate one set of steps for the first step, and another set for the second step
+	// because the texts are different after progressing from the first step to the second step
+	useEffect( () => {
+		const isInstalling =
+			state === JetpackPluginStates.INSTALLING || isWaitingForRedirect;
+		if ( step === 'first' ) {
+			setStepsToDisplay( [
+				{
+					key: 'first',
+					label: __( 'Connect to Jetpack', 'woocommerce' ),
+					description: __(
+						'To get started, install Jetpack - our free tool required to sync your store with the WooCommerce mobile app',
+						'woocommerce'
+					),
+					content: (
+						<>
+							<Button
+								className="install-jetpack-button"
+								onClick={ () => {
+									setIsWaitingForRedirect( true );
+									installHandler();
+								} }
+							>
+								{ isInstalling && (
+									<Spinner className="install-jetpack-spinner" />
+								) }
+								<div
+									style={ {
+										visibility: isInstalling
+											? 'hidden'
+											: 'visible',
+									} }
+									className="install-jetpack-button-contents"
+								>
+									<div className="jetpack-icon" />
+									<div className="install-jetpack-button-text">
+										{ __(
+											'Install and Connect',
+											'woocommerce'
+										) }
+									</div>
+								</div>
+							</Button>
+						</>
+					),
+				},
+				{
+					key: 'second',
+					label: 'Sign into the app',
+					description: '',
+					content: <div>Second step content.</div>,
```

Comment:
This looks like default content. Do we need it? 🤔

Reply:
I'm surprised this didn't show up in the modal, must've overflowed. Thanks for catching it! The key does need to be there as it's not optional in TS